distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso

```diff
@@ -878,16 +901,25 @@ type Vector a
          Sorting a vector of `Pair`s on the first element, descending.
 
              [Pair 1 2, Pair -1 8].sort Sort_Direction.Descending (_.first)
-    sort : Sort_Direction -> (Any -> Any) -> (Any -> Any -> Ordering) -> Vector Any ! Incomparable_Values
-    sort self (order = Sort_Direction.Ascending) (on = x -> x) (by = (Ordering.compare _ _)) =
-        comp_ascending l r  = by (on l) (on r)
-        comp_descending l r = by (on r) (on l)
-        compare = if order == Sort_Direction.Ascending then comp_ascending else
-            comp_descending
-
-        new_vec_arr = self.to_array.sort_builtin compare
-        if new_vec_arr.is_error then Error.throw new_vec_arr else
-            Vector.from_polyglot_array new_vec_arr
+
+       > Example
+         Sorting a vector with elements with different comparators. Values 1
+         and My_Type have different comparators. 1 will be sorted before My_Type
+         because it has the default comparator, and warning will be attached to
+         the resulting vector.
+
+             [My_Type.Value 'hello', 1].sort == [1, My_Type.Value 'hello']
+    sort : Sort_Direction -> (Any -> Any)|Nothing -> (Any -> Any -> (Ordering|Nothing))|Nothing -> Problem_Behavior -> Vector Any ! Incomparable_Values
+    sort self (order = Sort_Direction.Ascending) on=Nothing by=Nothing on_incomparable=Problem_Behavior.Ignore =
+        comps = case on == Nothing of
+            True -> self.map it-> Comparable.from it
+            False -> self.map it-> Comparable.from (on it)
+        compare_funcs = comps.map (it-> it.compare)
+        problem_behavior = case on_incomparable of
+            Problem_Behavior.Ignore -> 0
+            Problem_Behavior.Report_Warning -> 1
+            Problem_Behavior.Report_Error -> 2
```

Comment:
Ugh. This looks very meh. Although indeed I'm not sure what better way there is to do this without making `Problem_Behavior` a builtin. Maybe we just should make this Atom a builtin if it's used here??

If not, I'd at least move this mapping to some external method, probably in `Problem_Behavior.enso` as a helper, _not_ on the Problem_Behavior type but on module to keep it hidden. Ideally I'd use some constants coming from Java for these numbers. Maybe you can define an `enum` in the core polyglot utils to access it from both sides. Although this starts sounding like an overkill for sth so simple, so up to you.

But I'd at least split it off to a separate method - if we ever need this mapping in another place, I don't want it being copied-and-pasted, let's have a single method to reference.

Reply:
Refactored this into a private, documented, method `Problem_Behavior.to_number`.